src/Adapter/Order/OrderAmountUpdater.php

```diff
@@ -309,16 +309,31 @@ private function updateOrderTotals(Order $order, Cart $cart, int $computingPreci
         $order->total_shipping_tax_incl = $cart->getOrderTotal(true, Cart::ONLY_SHIPPING, $orderProducts, $carrierId, false, $this->keepOrderPrices);
 
         if (!$this->getOrderConfiguration('PS_ORDER_RECALCULATE_SHIPPING', $order)) {
-            $shippingDiffTaxIncluded = $order->total_shipping_tax_incl - $totalShippingTaxIncluded;
-            $shippingDiffTaxExcluded = $order->total_shipping_tax_excl - $totalShippingTaxExcluded;
+            $freeShipping = false;
+            foreach ($order->getCartRules() as $cartRule) {
+                if ($cartRule['free_shipping']) {
+                    $freeShipping = true;
+                    break;
+                }
+            }
```

Comment:
```suggestion
            $freeShipping = $this->isFreeShipping($order);
```

You can create a method
```php
    protected function isFreeShipping(Order $order): bool
    {
        foreach ($order->getCartRules() as $cartRule) {
            if ($cartRule['free_shipping']) {
                return true;
            }
        }

        return false;
    }
```

to prevent, duplicate